lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -224,8 +224,18 @@ static bool recursiveType(const StructType *ST, const Type *Ty) {
              StructTy->element_end();
     }
 
-    if (auto *PtrTy = dyn_cast<PointerType>(Ty))
-      return Run(PtrTy->getPointerElementType());
+    if (auto *PtrTy = dyn_cast<PointerType>(Ty)) {
+      auto *ElTy = PtrTy->getPointerElementType();
+      if (auto *FTy = dyn_cast<FunctionType>(ElTy)) {
+        // If we have a function pointer, then argument types and return type of
+        // the referenced function also need to be checked
+        return Run(FTy->getReturnType()) ||
+               find_if(FTy->param_begin(), FTy->param_end(), Run) !=
```

Comment:
Minor. Can we use `llvm::any_of` or `std::any_of` here ?

Reply:
Done